### PR TITLE
Adding support for Sharp GP2Y10 Dust sensor

### PR DESCRIPTION
**Pull request by remmeha**
## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on core 2.6.1
  - [ ] The code change pass travis tests. **Your PR cannot be merged unless tests pass**
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

**Comment by arendst** [comment]
Thx for you effort but...

All this for just reading an analog input?
Using a delay of 16 x 10 milliseconds while interrupts are disabled makes the device virtually unusable.

Don't think this is going to be implemented any time soon.

**Comment by remmeha** [comment]
Well. You have to pulse the device and exactly time when you read the analog input. That's why all the effort. It works for me. Thanks.

File: tasmota/language/en-GB.h [hunk 1]
```diff
@@ -628,6 +628,7 @@
 #define D_SENSOR_SLAVE_TX    "Slave TX"
 #define D_SENSOR_SLAVE_RX    "Slave RX"
 #define D_SENSOR_SLAVE_RESET "Slave RST"
+#define D_SENSOR_GP2Y10      "GP2Y10"
```
Comment (ascillato2) [inline tasmota/language/en-GB.h]: This key is missing for the rest of the Language files making travis test to fail.

**Comment by andrethomas2** [comment]
Closing this as it cannot be merged in its current form.

**Comment by remmeha** [comment]
I was working on a fix for the language files as well as another method of averaging which makes the interrupt disable only 3ms. Do you it is interesting? Shall I make a new pull request?

**Comment by ascillato** [comment]
https://github.com/arendst/Tasmota/pull/7118

**Pull request closed without merging**